doc/sai-ptf/warmboot.md

```diff
@@ -0,0 +1,262 @@
+   ```shell
+   sai_warmboot.sh
+   ```
+1. change the profile, enable the warm start
+   ```shell
+   # Sample in a dell s6000
+   cat /etc/sai.d/sai.profile
+   SAI_INIT_CONFIG_FILE=/usr/share/sonic/hwsku/td2-s6000-32x40G.config.bcm
+   SAI_NUM_ECMP_MEMBERS=32
+   SAI_WARM_BOOT_WRITE_FILE=/var/warmboot/sai-warmboot.bin
+   SAI_WARM_BOOT_READ_FILE=/var/warmboot/sai-warmboot.bin
+   SAI_BOOT_TYPE=1
+   ```
+
+1. trigger the test in starting mode
+   ```shell
+   ptf --test-dir /tmp/test/SAI/ptf/warm_boot warm_saiswitch.WarmAvailableIPv4RouteEntryTest --interface '0-0@eth0' ... '0-31@eth31' --relax "--test-params=thrift_server='Ip_address';test_reboot_mode='warm';test_reboot_stage='starting'"
+   ```
+   *p.s. Need to add the parameters test_reboot_mode='warm';test_reboot_stage='starting'*
+
+### Post warmboot - After warm reboot
+1. start saiserver and trigger the test in post mode
+   ```shell
+   ptf --test-dir /tmp/test/SAI/ptf/warm_boot warm_saiswitch.WarmAvailableIPv4RouteEntryTest --interface '0-0@eth0' ... '0-31@eth31' --relax "--test-params=thrift_server='Ip_address';test_reboot_mode='warm';test_reboot_stage='post'"
+   ```
+   *p.s. Need to add the parameters test_reboot_mode='warm';test_reboot_stage='post'*
+
+
+## Example of re-organizing the case into a warm reboot
+### Context
+For warm reboot test cases, we can always make it in two ways:
+1. Reuse the test logic in existing code, in this way we need to:
+     - wrap the code in the existing method for warm reboot usage, and
+     - place the new method in the warm reboot method
+2. Write a brand new warm reboot test cases
+
+In order to reuse the existing test cases in a warm reboot, we need to re-organize the existing test cases into the warm reboot structure.
+
+### Test case detail
+There is a test sample for the warm reboot test https://github.com/opencomputeproject/SAI/pull/1440
+
+
+For this warm reboot test in the sample, we reuse the code in the [L2SanityTest](https://github.com/richardyu-ms/SAI/blob/v1.7/ptf/saisanity.py). In general, in this test case, we 
+- Setup
+  1. create VLANs for each port
+  2. create MACs for ports (mapped with index)
+  3. then create route rules by adding mac and port
+  4. then send packets to on port by with different mac
+- Check
+
+  5. check we can get the port allocated with the mac as route rules
+
+With that basic VLAN and FDB functionality, we expect in a warm reboot scenario, we can get the test case and expectations below
+1. Pre-warm-reboot
+    - Setup as a test in L2SanityTest
+    - Check as a test in L2SanityTest
+2. Starting
+    - Check as a test in L2SanityTest
+
+*In this stage, like a switch running in starting mode, sai switch is not getting started, we will only verify the packet transit*
+    
+3. Post-warm-reboot
+    - APIs start the switch in warm mode
+    - Check as a test in L2SanityTest
+
+*In this stage, the switch gets started, and we can use warm reboot API to start the switch.*
```

Comment:
Can we elaborate in the doc what exactly is being verified after warmboot?  Also can we add reconciliation /ensuring that SW entries are same as HW entries (and not just by using packets)?

Reply:
added a conclusion in the end